crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -2468,16 +2536,62 @@ impl AuthorityPerEpochStore {
                     })
                     .collect();
 
+                // As long as we are still accepting consensus certs, each commit also needs
+                // to generate a new round of randomness.
+                if let Some(randomness_obj_initial_shared_version) = self
+                    .epoch_start_config()
+                    .randomness_obj_initial_shared_version()
+                {
+                    if self.protocol_config().random_beacon()
+                        && self
+                            .get_reconfig_state_read_lock_guard()
+                            .should_accept_consensus_certs()
```

Comment:
Please add comment on why it is safe to release the lock immediately.

Reply:
done